File: app/src/main/java/com/agrogames/islandsofwar/map/impl/Water.java
package com.agrogames.islandsofwar.map.impl;

import com.agrogames.islandsofwar.engine.abs.common.Cell;
import com.agrogames.islandsofwar.engine.abs.common.Point;
import com.agrogames.islandsofwar.engine.abs.map.MapObject;

public class Water implements MapObject {
    private final Cell location;

    public Water(Cell location) {
        this.location = location;
    }

    @Override
    public Cell[] GetTerritory() {
        return new Cell[]{location};
    }
}

File: app/src/main/java/com/agrogames/islandsofwar/map/impl/Water.kt
package com.agrogames.islandsofwar.map.impl

import com.agrogames.islandsofwar.engine.abs.common.Cell
import com.agrogames.islandsofwar.engine.abs.map.MapObject

class Water(private val location: Cell) : MapObject {
    override val territory: Array<Cell>
        get() = arrayOf(location)
    override val isMoving: Boolean
        get() = false
    override val height: Int
        get() = 0
}